modules/vault/src/main/java/org/apache/ignite/internal/vault/common/VaultListener.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.vault.common;
+
+/**
+ * Vault storage listener for changes.
+ */
+@FunctionalInterface
+public interface VaultListener {
+    /**
+     * Method called when entries in storage change.
+     * @param changedEntry Changed value
+     */
+    void onEntryChanged(VaultEntry changedEntry);
```

Comment:
Just `onUpdate`? And update could be batch, so it will be good idea to get `Iterable<Entry>`. 
Also, how you will reject update listener? It seems this method should return `boolean`.